### PR TITLE
utmpx api for linux musl + musl few build fixes.

### LINKED ISSUE #3190: utmp function family missing on musl
Hello,
functions like getutxent, setutxent, endutxent, struct utmpx, etc. are missing from the musl implementation ( src/unix/linux_like/linux/musl/mod.rs ), even though those functions are implemented in musl (see https://git.musl-libc.org/cgit/musl/tree/include/utmpx.h ).

As reference, see this [commit](https://github.com/rust-lang/libc/commit/98fd3188e7d46db84d321475344f2f82d958b0c7) which added the function for solaris, as well as the man page: https://linux.die.net/man/3/endutxent

### PR DESCRIPTION
**Pull request by devnexen**
close #3190

### CODE REVIEW

**Comment by joshtriplett** [comment]
Why is this trying to change O_LARGEFILE?

**Comment by devnexen** [comment]
I just looked at an alpine instance, O_LARGEFILE on x86_64 arch is the same as android so `0o100000`

**Comment by Ecordonnier** [comment]
When can this PR be merged? Is it blocked waiting for a future release?

**Comment by devnexen** [comment]
@joshtriplett did I answer your question ?
@JohnTitor would you mind reviewing it ?
thanks.

**Comment by Ecordonnier** [comment]
@JohnTitor can this be merged? What does the tag "breakage-candidate" mean?

**Comment by JohnTitor** [comment]
Turned out that the Cargo feature way I mentioned above isn't pragmatic, and we have to make libc 0.3 real. See https://github.com/rust-lang/libc/pull/3488#issuecomment-1862579094 for more details.

File: src/unix/linux_like/linux/musl/mod.rs [hunk 1]
```diff
@@ -779,6 +779,13 @@ extern "C" {
 
     pub fn dirname(path: *mut ::c_char) -> *mut ::c_char;
     pub fn basename(path: *mut ::c_char) -> *mut ::c_char;
+
+    pub fn getutxent() -> *mut utmpx;
+    pub fn getutxid(ut: *const utmpx) -> *mut utmpx;
+    pub fn getutxline(ut: *const utmpx) -> *mut utmpx;
+    pub fn pututxline(ut: *const utmpx) -> *mut utmpx;
+    pub fn setutxent();
+    pub fn endutxent();
```
Comment (JohnTitor) [inline src/unix/linux_like/linux/musl/mod.rs:788]: For now, we can merge non-breaking changes like this.

**Review by JohnTitor — APPROVED** [review]

**Comment by Ecordonnier** [comment]
Now that this has been backported to the 0.2 branch, I tested the build of uutils-coreutils with the libc crate and musl using this commit, however FYI due to the lack of the utmpxname function, as well as the symbols "libc::__UT_HOSTSIZE", "libc::__UT_LINESIZE", "libc::__UT_NAMESIZE" and "libc::ACCOUNTING", I wasn't able to compile uutils-coreutils.

**Comment by tgross35** [comment]
@Ecordonnier If you are interested in putting up a PR, though should be reasonably trivial to add. Otherwise just copy that to a new issue, somebody will likely pick it up reasonably fast.

**Comment by Ecordonnier** [comment]
@tgross35 I created https://github.com/rust-lang/libc/issues/4322 as follow-up issue. I'll try to work on this if I find some time.

**Pull request merged**